addons/l10n_it_edi/models/account_edi_proxy_user.py

```diff
@@ -25,3 +30,202 @@ def _compute_proxy_type(self):
         for user in self:
             if user.company_id.country_code == 'IT':
                 user.proxy_type = 'l10n_it_edi'
+
+    def _get_proxy_identification(self, company, proxy_type):
+        if proxy_type == 'l10n_it_edi':
+            if not company.l10n_it_codice_fiscale:
+                raise UserError(_('Please fill your codice fiscale to be able to receive invoices from FatturaPA'))
+            return self.env['res.partner']._l10n_it_edi_normalized_codice_fiscale(company.l10n_it_codice_fiscale)
```

Comment:
```suggestion
            return company.partner_id._l10n_it_edi_get_normalized_codice_fiscale()
```
```py
    def _l10n_it_edi_normalized_codice_fiscale(self):
        """ Normalize the Italian Tax Code for export.
            If the Tax Code is equal to the Italian VAT, it may mistakenly have the country prefix,
            so we try and remove it if we can
        """
        self.ensure_one()
        if self.l10n_it_codice_fiscale and re.match(r'^IT[0-9]{11}$', self.l10n_it_codice_fiscale):
            return self.l10n_it_codice_fiscale[2:13]
        return self.l10n_it_codice_fiscale
```

Reply:
Note that we can't change the `_l10n_it_edi_normalized_codice_fiscale` method as you suggest since we need it at https://github.com/odoo/odoo/blob/a83ea61434784d23aeb616badf20232fba98d553/addons/l10n_it_edi/models/res_partner.py#L141